ui/react-components/components/AddAppointment/AddAppointment.jsx

```diff
@@ -208,12 +230,30 @@ const AddAppointment = props => {
         return {...prevAppointmentDetails, ...modifiedAppointmentDetails}
     });
 
+    const isDatelessAppointment = () => {
+        return appointmentDetails && appConfig && appConfig.prioritiesForDateless.includes(appointmentDetails.priority)
+            && !appointmentDetails.appointmentDate && !appointmentDetails.startTime && !appointmentDetails.endTime;
+    }
+
     const isValidAppointment = () => {
         const isValidPatient = appointmentDetails.patient && appointmentDetails.patient.value.uuid;
-        const startTimeBeforeEndTime = isStartTimeBeforeEndTime(appointmentDetails.startTime, appointmentDetails.endTime);
-        updateCommonErrorIndicators(isValidPatient, startTimeBeforeEndTime);
-        updateErrorIndicators({appointmentDateError: !appointmentDetails.appointmentDate});
-        return isValidPatient && appointmentDetails.service && appointmentDetails.appointmentDate && appointmentDetails.startTime && appointmentDetails.endTime && startTimeBeforeEndTime;
+        const isValidPriority = !isAppointmentPriorityOptionEnabled(appConfig) ? true : ((isAppointmentPriorityOptionEnabled(appConfig) && !appointmentDetails.priority) ? false : true)
```

Comment:
This is not scalable and can be simplified a lot more. 
1. isAppointmentPriorityOptionEnabled(appConfig) is being called twice for alternate uses within this statement
2. Returning true/false from a conditional is an anti-pattern.
3. In fact, this is very convoluted since you have inverse condition at the second ternary and are returning a false in case the check succeeds. Negative expressions like this are very unreadable

Better approach:
const isValidProperty = !isAppointmentPriorityOptionEnabled(appConfig) || appointmentDetails.priority

I believe this would lead to the same answer

Reply:
updated